### DISCUSSION TITLE
XMLHttpRequest caching support

### DISCUSSION THREAD

**Discussion by jexroid** [vercel/next.js#66844] [Ideas]
### Goals

1. Ability to choose how **XHR** request should be cached
same like the `fetch()` method which extends the cache property

2. Revalidating the **XHR** request.
ability to set the revalidation time for the XHR request

3. Ability to Tag the Cache and revalidate it using an on-demand revalidating

### Background

- ### Why do we need to use **XHR** request?
I know you might think `XML HTTP Request` is an old javascript API that is now deprecated because of synchronous usage, **BUT** it's still in use because of these features:
1. onprogress() method
you can use onprogress method to monitor your upload to the server calculate when your packet will be ended completely to the server, and track your upload based on percentage.
2. Eventlistener
Ability to execute a function and react to Events like: `abort | error | load | loadend | loadstart | progress | readystatechange | timeout`
2. Regex
you can use Regex to scan the XML content

these are just programming capabilities that XMLHttpRequest has. but there is far more power in **XHR** that can bring Nextjs to a better framework.

- ### In the modern web, When XML HTTP Request is used?
XMLHttpRequest (XHR) objects are used to interact with servers. You can retrieve data from a URL without doing a full page refresh. but the main is to update just part of a page without disrupting what the user is doing or refreshing the whole page. [link](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)

**in most cases XHR is used in File Uploads or the places where monitoring the data without disruption is essential.**

### Proposal

I'm interested in contributing but I feel XHR is being used in Nextjs but it needs to become something greater for developers to use and I see a potential for making Tools from it, unfortunately, I may not be an expert in React, but I am well-versed in the workings of HTTP and love to hear from you.

**Comment by SouthLink** [comment]
You can use [Route Segment Config](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config) to control the cache strategy for the entire route segment, which is the same as extending the fetch cache，And applies to other third-party request libraries，
[[Fetching data on the Server with third-party libraries](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-third-party-libraries)](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-third-party-libraries) 。

**Reply by jexroid** [reply]
thanks. I had a feeling nextjs has this feature
